### PR TITLE
3337 Add Refname and Refby in SRS table of abbreviations

### LINKED ISSUE #3337: “RefBy” and “Refname” should be abbreviations in the SRS Table of Abbreviations
We currently use the two abbreviations in the SRS documents, but I haven't found any occurrence of them in the table of abbreviations and acronyms.

### PR DESCRIPTION
**Pull request by harmanpreet-sagar**
Closes #3337: “RefBy” and “Refname” should be abbreviations in the SRS Table of Abbreviations. Changes have been made to stable, example, and the Documentation code.

### CODE REVIEW

**Comment by smiths** [comment]
The changes to stable look fine.  I'm not sure why the build failed.

**Comment by harmanpreet-sagar** [comment]
Fixed errors.

**Comment by balacij** [comment]
Ah, interesting catch @samm82! I think we should almost definitely be doing that. I think @JacquesCarette previously mentioned that all 'Show' instances should be limited to debugging instead of usage. I suppose it is because of what you're mentioning now -- specifically that they don't have a way of resolving information from the chunk database.

This PR looks good to me!  Are we able to use these new `CI`s in the SRS printer? We might need to check out `drasil-docLang` for all instances of `RefBy` and `Refname`, replacing them all with usage of this `CI`. (To be clear, if we can, we should continue working on this PR to use this `CI` appropriately too.)

I think those `acronyms` lists might be movable to a shared file for creating all SRS documents. This would help with #3326.

**Comment by balacij** [comment]
For sure. It shouldn't be done in this PR.

**Review by samm82 — APPROVED** [review]
I've created a new issue for the discussed issue (#3393), so this looks good to me!

File: code/drasil-data/lib/Data/Drasil/Concepts/Documentation.hs [hunk 1]
```diff
@@ -75,6 +75,8 @@ typUnc      = commonIdeaWithDict "typUnc"      (cn' "typical uncertainty")
 sec         = commonIdeaWithDict "section"     (cn' "section")                                       "Sec"     [documentc]
 srs         = commonIdeaWithDict "srs"         softReqSpec                                           "SRS"     [softEng]
 notebook    = commonIdeaWithDict "notebook"    (cn' "notebook")                                      "NB"      [softEng]
+refBy       = commonIdeaWithDict "refBy"       (cn' "Referenced By")                                 "RefBy"   [documentc]
```
Comment (JacquesCarette) [inline code/drasil-data/lib/Data/Drasil/Concepts/Documentation.hs:78]: One thing to change: we put our concepts all in lower-case, and then Drasil takes care of casing if/when needed.

**Review by JacquesCarette — CHANGES_REQUESTED** [review]

File: code/drasil-data/lib/Data/Drasil/Concepts/Documentation.hs (hunk 1 above)
Comment (harmanpreet-sagar) [reply code/drasil-data/lib/Data/Drasil/Concepts/Documentation.hs:78]: Thanks for pointing that out! I will make the change now.

File: code/stable/projectile/SRS/JSON/Projectile_SRS.ipynb [hunk 2]
```diff
@@ -127,7 +127,7 @@
     "|IM|Instance Model|\n",
     "|PS|Physical System Description|\n",
     "|R|Requirement|\n",
-    "|RefBy|Referenced By|\n",
+    "|RefBy|Referenced by|\n",
     "|Refname|Reference Name|\n",
```
Comment (samm82) [inline code/stable/projectile/SRS/JSON/Projectile_SRS.ipynb:131]: Why does "Referenced by" have a lowercase "b" but "Reference Name" have an uppercase "N"?

Comment (balacij) [reply code/stable/projectile/SRS/JSON/Projectile_SRS.ipynb:131]: I _think_ that displaying in the final artifacts should have the capital letters, it's just that the 'backend' code needs to use the lowercase copies.

Comment (samm82) [reply code/stable/projectile/SRS/JSON/Projectile_SRS.ipynb:131]: Right, but this is in a generated artifact (the Jupyter notebook for Projectile), so we would expect a capital "B" in "Referenced By"

Comment (balacij) [reply code/stable/projectile/SRS/JSON/Projectile_SRS.ipynb:131]: Yep, exactly :smile:

Comment (harmanpreet-sagar) [reply code/stable/projectile/SRS/JSON/Projectile_SRS.ipynb:131]: I agree. It should be "Referenced By" with a capital B. I will try and fix that issue right now!

File: code/drasil-lang/lib/Language/Drasil/NounPhrase.hs [hunk 3]
```diff
@@ -270,5 +270,5 @@ capWords w@(c:cs)
 
 -- | Words that should not be capitalized in a title (prepositions, articles, or conjunctions).
 doNotCaps :: [String]
-doNotCaps = ["a", "an", "the", "at", "by", "for", "in", "of",
+doNotCaps = ["a", "an", "the", "at", "for", "in", "of",
```
Comment (samm82) [inline code/drasil-lang/lib/Language/Drasil/NounPhrase.hs:273]: Hmm... if this is the only solution to capitalizing "By", it's probably worth reverting this and making the capitalization of "By" a separate issue. We would (likely) still want "by" to be lowercase in a normal title, but uppercase in our specific use of "Referenced By" in the table, so we would need some more design to figure out how to do that.

**Review by samm82 — CHANGES_REQUESTED** [review]

File: code/drasil-lang/lib/Language/Drasil/NounPhrase.hs (hunk 3 above)
Comment (JacquesCarette) [reply code/drasil-lang/lib/Language/Drasil/NounPhrase.hs:273]: Agreed!

Comment (harmanpreet-sagar) [reply code/drasil-lang/lib/Language/Drasil/NounPhrase.hs:273]: Ok, I will revert the change and create a separate issue for that.

File: code/drasil-data/lib/Data/Drasil/Concepts/Documentation.hs [hunk 4]
```diff
@@ -318,19 +318,17 @@ vavPlan                      = compoundNC vav plan
...
+funcReqDom    = ccs (mkIdea "funcReqDom"    (functionalRequirement ^. term)    $ Just "FR")  EmptyS [reqDom]
+nonFuncReqDom = ccs (mkIdea "nonFuncReqDom" (nonfunctionalRequirement ^. term) $ Just "NFR") EmptyS [reqDom]
+chgProbDom    = ccs (nc "chgProbDom" $ cn' "change")                                         EmptyS [srsDom]
+likeChgDom    = ccs (mkIdea "likeChgDom"    (likelyChg ^. term)                $ Just "LC")  EmptyS [chgProbDom]
+unlikeChgDom  = ccs (mkIdea "unlikeChgDom"  (unlikelyChg ^. term)              $ Just "UC")  EmptyS [chgProbDom]
 
 -- | List of SRS-related concepts, including SRS.
 srsDomains :: [ConceptChunk]
-srsDomains = [cw srsDom, goalStmtDom, reqDom, funcReqDom, nonFuncReqDom, assumpDom, chgProbDom, likeChgDom, unlikeChgDom, refByDom, refNameDom]
+srsDomains = [cw srsDom, goalStmtDom, reqDom, funcReqDom, nonFuncReqDom, assumpDom, chgProbDom, likeChgDom, unlikeChgDom]
```
Comment (samm82) [inline code/drasil-data/lib/Data/Drasil/Concepts/Documentation.hs]: Were the changes from this file also meant to be reverted?

Comment (harmanpreet-sagar) [reply code/drasil-data/lib/Data/Drasil/Concepts/Documentation.hs]: No, but I was not sure how to revert only specific changes, so I reverted the entire commit and am creating a new commit to add those changes.

Comment (samm82) [reply code/drasil-data/lib/Data/Drasil/Concepts/Documentation.hs]: A good lesson in making small commits, but I also know the difficulty in doing so when it takes many changes to get a result you're happy with committing 😅

Comment (harmanpreet-sagar) [reply code/drasil-data/lib/Data/Drasil/Concepts/Documentation.hs]: Yeah, I did not want to commit anything that would necessarily break the PR! But most definitely a lesson that I will make sure I use from now on.

File: code/drasil-data/lib/Data/Drasil/Concepts/Documentation.hs [hunk 5]
```diff
@@ -75,6 +75,8 @@ typUnc      = commonIdeaWithDict "typUnc"      (cn' "typical uncertainty")
 sec         = commonIdeaWithDict "section"     (cn' "section")                                       "Sec"     [documentc]
 srs         = commonIdeaWithDict "srs"         softReqSpec                                           "SRS"     [softEng]
 notebook    = commonIdeaWithDict "notebook"    (cn' "notebook")                                      "NB"      [softEng]
+refBy       = commonIdeaWithDict "refBy"       (cn "referenced by")                                  "RefBy"   [documentc]
```
Comment (samm82) [inline code/drasil-data/lib/Data/Drasil/Concepts/Documentation.hs:78]: SUPER minor nitpick is to just add a space to help align this "dabatase" table
```suggestion
refBy       = commonIdeaWithDict "refBy"       (cn  "referenced by")                                 "RefBy"   [documentc]
```

**Review by samm82 — CHANGES_REQUESTED** [review]
Other than that minor formatting thing, this looks great! It's a shame that the "By" issue was so complicated, but I'm sure we'll figure that out soon enough 😉

File: code/drasil-data/lib/Data/Drasil/Concepts/Documentation.hs (hunk 5 above)
Comment (harmanpreet-sagar) [reply code/drasil-data/lib/Data/Drasil/Concepts/Documentation.hs]: Thanks for catching! Fixed it now.

**Review by samm82 — APPROVED** [review]
Awesome!

**Review by JacquesCarette — APPROVED** [review]

**Comment by balacij** [comment]
> @balacij From what I can tell, "Refname" isn't used in the SRS, and this is the only usage of "RefBy". This is the only instance of "RefBy" I could find in the repo, which means this `Show` instance is used in usage, which may very well be a code smell, one that may be complex enough to merit its own issue/PR.

Just following up on this: we should try to remove the `Show` instance and see what breaks, and then replace it with usage of this `CI`. Is there a ticket following this already? @harmanpreet-sagar @samm82

**Pull request merged**